src/phoenix/server/api/schema.py

```diff
@@ -343,6 +347,42 @@ async def clear_project(self, info: Info[Context, None], id: GlobalID) -> Query:
                 cache.invalidate(ClearProjectSpansEvent(project_rowid=project_id))
         return Query()
 
+    @strawberry.mutation
+    async def create_dataset(
+        self,
+        info: Info[Context, None],
+        name: str,
+        description: Optional[str],
+        metadata: Optional[JSON],
+    ) -> Dataset:
+        metadata = metadata or {}
+        async with info.context.db() as session:
+            result = await session.execute(
+                insert(models.Dataset)
+                .values(
+                    name=name,
+                    description=description,
+                    metadata_=metadata,
+                )
+                .returning(
+                    models.Dataset.id,
+                    models.Dataset.created_at,
+                    models.Dataset.updated_at,
+                )
+            )
+            row = result.fetchone()
```

Comment:
Do we need to fetch here, or can we just return the entire `model.Dataset` in `returning` and convert that to the GraphQL type?

Reply:
I don't actually know to do that